Zend/tests/property_hooks/readonly_rfc_example_lazy_product.phpt

```diff
@@ -0,0 +1,86 @@
+--TEST--
+Readonly classes can be constructed via reflection by ORM
+--FILE--
+<?php
+
+interface DbConnection {
+   public function loadCategory(string $id): Category;
+}
+
+class Category {
+   public function __construct(public string $name) {}
+}
+
+class MockDbConnection implements DbConnection {
+   public function loadCategory(string $id): Category {
+       echo "hit database\n";
+       return new Category("Category {$id}");
+   }
+}
+
+readonly class Product
+{
+    public function __construct(
+        public string $name,
+        public float $price,
+        public Category $category,
+    ) {}
+}
+
+readonly class LazyProduct extends Product
+{
+   private DbConnection $dbApi;
+
+   private string $categoryId;
+
+   public Category $category {
+       get {
+           return $this->category ??= $this->dbApi->loadCategory($this->categoryId);
+       }
+   }
+}
```

Comment:
```suggestion

class Category {
   public function __construct(public string $name) {}
}

readonly class Product
{
    public function __construct(
        public Category $category,
    ) {}
}

readonly class LazyProduct extends Product
{
   private string $categoryId;

   public Category $category {
       get {
           return $this->category ??= new Category($this->categoryId);
       }
   }
}
```

Something like this should still get the point across.  (The reflection checks below will need to be updated as well.)